internal/moduletest/graph/variables.go

```diff
@@ -53,13 +53,18 @@ func (n *NodeTestRun) GetVariables(ctx *EvalContext, includeWarnings bool) (terr
 
 	// First, let's step through the expressions within the run block and work
 	// them out.
+
 	for name, expr := range run.Config.Variables {
 		requiredValues := make(map[string]cty.Value)
 
 		refs, refDiags := langrefs.ReferencesInExpr(addrs.ParseRefFromTestingScope, expr)
 		for _, ref := range refs {
 			if addr, ok := ref.Subject.(addrs.InputVariable); ok {
-				if value, valueDiags := ctx.VariableCache.GetVariableValue(addr.Name); value != nil {
+				if value := ctx.GetVariable(addr.Name); value != nil {
+					requiredValues[addr.Name] = value.Value
+					continue
+				}
+				if value, valueDiags := ctx.EvaluateUnparsedVariable(addr.Name, run.ModuleConfig.Module.Variables[addr.Name]); value != nil {
```

Comment:
Same question I asked earlier. Shouldn't these references have been resolved before we get here?